test/widgets/lightbox_test.dart

```diff
@@ -271,9 +298,97 @@ void main() {
+    //     thumbnailUrl: thumbnail,
+    //     originalWidth: 1000,
+    //     originalHeight: 1000);
+
+    //   // Both visible
+    //   tester.widget(find.byWidgetPredicate(
+    //     (widget) => widget is RealmContentNetworkImage && widget.src == thumbnail));
+    //   tester.widget(find.byWidgetPredicate(
+    //     (widget) => widget is RealmContentNetworkImage && widget.src == src));
+
+    //   print('pump');
+    //   await tester.pump();
+    //   print('done pump');
+    //   tester.widgetList<RealmContentNetworkImage>(find.byType(RealmContentNetworkImage)).forEach((w) => print(w.src));
+
+    //   print('pump');
+    //   await tester.pump();
+    //   print('done pump');
+    //   tester.widgetList<RealmContentNetworkImage>(find.byType(RealmContentNetworkImage)).forEach((w) => print(w.src));
+
+    //   // TODO verify only src visible
+
+    //   debugNetworkImageHttpClientProvider = null;
+    //   PaintingBinding.instance.imageCache.clear();
+    // });
+
+    testWidgets('image is scaled down to fit', (tester) async {
+      tester.view.physicalSize = const Size(800, 600);
+      tester.view.devicePixelRatio = 1.0;
+      prepareBoringImageHttpClient(content: kSolidBlueLargeAvatar);
+      final message = eg.streamMessage(sender: eg.otherUser);
+      await setupPage(tester, message: message, thumbnailUrl: null);
+
+      // Ensure only src image is visible
+      tester.widgetList(find.descendant(
+        of: find.byType(RealmContentNetworkImage),
+        matching: find.byType(RawImage)),
+      ).single;
+
+      // Image is 1000x1000, but widget should fit the view size.
+      final renderImage = tester.renderObject<RenderImage>(find.descendant(
+        of: find.byType(RealmContentNetworkImage),
+        matching: find.byType(RawImage)));
+      check(renderImage.size).equals(tester.view.physicalSize);
+
+      debugNetworkImageHttpClientProvider = null;
+      PaintingBinding.instance.imageCache.clear();
     });
 
-    // TODO test _CopyLinkButton
+    // FAILS
+    // testWidgets('image is smaller than view size, it doesn\'t scale up', (tester) async {
+    //   tester.view.physicalSize = const Size(800, 600);
+    //   tester.view.devicePixelRatio = 1.0;
+    //   prepareBoringImageHttpClient(content: kSolidBlueAvatar);
+    //   final message = eg.streamMessage(sender: eg.otherUser);
+    //   await setupPage(tester, message: message, thumbnailUrl: null);
+
+    //   // Ensure only src image is visible
+    //   tester.widgetList(find.descendant(
+    //     of: find.byType(RealmContentNetworkImage),
+    //     matching: find.byType(RawImage)),
+    //   ).single;
+
+    //   // Image is 100x100, it should stay that size and shouldn't scale up.
+    //   final renderImage = tester.renderObject<RenderImage>(find.descendant(
+    //     of: find.byType(RealmContentNetworkImage),
+    //     matching: find.byType(RawImage)));
+    //   check(renderImage.size).equals(const Size(100, 100));
```

Comment:
I can't get this test to pass because seems like flutter test enviroment doesn't report correct image sizes?